src/Uno.UI.Runtime.Skia.Wpf/WpfApplicationExtension.cs

```diff
@@ -1,20 +1,75 @@
 ﻿using System;
+using Microsoft.Win32;
+using Uno.Helpers.Theming;
+using Uno.UI.Runtime.Skia.Wpf.WPF.Extensions.Helper.Theming;
 using Uno.UI.Xaml;
 using Windows.UI.Xaml;
 
 namespace Uno.UI.Runtime.Skia.Wpf
 {
-	public class WpfApplicationExtension : IApplicationExtension
+	public class WpfApplicationExtension : IApplicationExtension, IDisposable
 	{
 		private readonly Application _owner;
+		private WpfSystemThemeHelperExtension _themeHelper;
+		private SystemTheme _currentTheme;
+
+		private bool _disposedValue;
 
 		public WpfApplicationExtension(Application owner)
 		{
 			_owner = owner ?? throw new ArgumentNullException(nameof(owner));
+
+			_themeHelper = new WpfSystemThemeHelperExtension(null);
+			_currentTheme = _themeHelper.GetSystemTheme();
+
+			SystemEvents.UserPreferenceChanged += SystemEvents_UserPreferenceChanged;
 		}
 
 		public bool CanExit => true;
 
 		public void Exit() => System.Windows.Application.Current.Shutdown();
+
+		public event EventHandler SystemThemeChanged;
+
+		private void SystemEvents_UserPreferenceChanged(object sender, UserPreferenceChangedEventArgs e)
+		{
+			// This event is raised when the theme is changed,
+			// but also for various other irrelevant reasons.
+			// Check the registry first, before forcing the application
+			// to refresh everything.
+			var _newTheme = _themeHelper.GetSystemTheme();
```

Comment:
Remove the underscore from local variable name.

Btw. useful trick! My system theme helper will fix this and make sure to prevent refresh if the theme does not actually change, but for now this will definitely help.

Reply:
Thanks, I've fixed the styling.

As for the "hack", there is another way that prevents false alarms: Watching the registry key.
But this requires really long coding and a lot of P/Invoke so I choose this method instead.